File: CSharp/Tests/CharInstancesInStringTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class CharInstancesInStringTest
    {
        [Theory]
        [InlineData('a', "edabit", 1)]
        [InlineData('b', "big fat bubble", 4)]
        [InlineData('c', "Chamber of secrets", 1)]
        [InlineData('f', "frank and his friends have offered five foxes for sale", 7)]
        [InlineData('x', "edabit", 0)]
        [InlineData('a', "Adam and Eve bit the apple and found a snake", 6)]
        [InlineData('s', "sssssssssssssssssssssssss", 25)]
        [InlineData('7', "10795426697", 2)]
        public void CharCount_StringAndCharInputs_ReturnCountOfCharacterInString(char myChar, string str, int expected)
        {
            var actual = CharInstancesInString.CharCount(myChar, str);

            Assert.Equal(expected, actual);
        }
    }
}

File: js/tests/charInstancesInString.test.js
const charCount = require("../charInstancesInString");

const data = [
  ["a", "edabit", 1],
  ["b", "big fat bubble", 4],
  ["c", "Chamber of secrets", 1],
  ["f", "frank and his friends have offered five foxes for sale", 7],
  ["x", "edabit", 0],
  ["a", "Adam and Eve bit the apple and found a snake", 6],
  ["s", "sssssssssssssssssssssssss", 25],
  ["7", "10795426697", 2]
];

describe("charCount", () => {
  describe("Return number of times the character is found in string", () => {
    test.each(data)("charCount(%p, %p) = %i", (myChar, str, result) => {
      var actual = charCount(myChar, str);

      expect(actual).toEqual(result);
    });
  });
});
